tools/codegen/src/main.rs

```diff
@@ -287,19 +285,40 @@ fn main() -> Result<()> {
                 }
             };
 
-            eprint!("downloading {url} for checksum ... ");
+            eprint!("downloading {url} for etag and checksum ... ");
             let download_cache = &download_cache_dir.join(format!(
                 "{version}-{platform:?}-{}",
                 Path::new(&url).file_name().unwrap().to_str().unwrap()
             ));
+            let response = download(&url)?;
+            let etag = response
+                .header("etag")
+                .expect("binary should have an etag")
+                .to_string()
+                .replace('\"', "");
+
+            if let Some(ManifestRef::Real(ref manifest)) = existing_manifest {
+                if let Some(entry) = manifest.download_info.get(&platform) {
+                    if entry.etag == etag {
+                        eprintln!("existing etag matched");
+                        download_info.insert(platform, entry.clone());
+                        continue;
+                    }
+                    eprintln!("warn: existing etag no longer valid.");
```

Comment:
probably?
I would like to understand github's assets etag a bit more before doing that.
While the etag being the same guarantees that the content hasnt changed,
If the etag changes, that has no specified meaning.
And now that I have downloaded all of the files, I don't want the tool silently deleting them all because github issued new etags for the same files.
If you agree, I can create an issue to run a full download of all assets again in a month and see if any etag changed.

Also if an etag changes, I would like to keep the old file, so I can inspect the old and new files.  Getting the old file might be difficult; the saved file on my disk might be the only way I could investigate why the github project uploaded a new asset easily by comparing the old and new.

i.e. I would prefer this warning to be replaced with a hard exit, but I didn't want to do that immediately.